app/cluster/dynamic_test.go

```diff
@@ -167,6 +186,22 @@ func TestDynamicCluster(t *testing.T) {
 		require.True(t, errorDB.callOrder > router.callOrder)
 	})
 
+	t.Run("Update workspaceIDs", func(t *testing.T) {
+		chACK := make(chan bool)
```

Comment:
```suggestion
		chACK := make(chan struct{})
```

By doing `bool` you're implicitly telling the reader that the value matters whereas you are just using the channel to signal here.

Reply:
Minor, but it makes sense.